src/classes/BDI_DataImportBatch_TDTM.cls

```diff
@@ -0,0 +1,70 @@
+/*
+    Copyright (c) 2020 Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Batch Data Import
+* @group-content ../../ApexDocContent/BatchDataImport.htm
+* @description Trigger Handler for the DataImportBatch SObject
+*/
+public class BDI_DataImportBatch_TDTM extends TDTM_Runnable {
+    
+    public static final String BATCH_STATUS_OPEN = 'Open';
+    public static final String BATCH_STATUS_COMPLETED = 'Completed';
+    public static final String BATCH_STATUS_FAILED = 'Failed - Needs Review';
```

Comment:
Could/should these statuses be exposed as custom labels so they can be translated?

Reply:
We could make these strings to be labels, however the di statuses were already stored as strings as well, see BDI_DataImport_API.cls (global static final string bdiImported = 'Imported'; etc). We should be consistent on both fronts I think. I am open to making all the statuses labels, but that would be a bigger refactoring as they are referenced elsewhere in the bdi code. @kenlewis-sfdo 
cc: @ptewson-sfdo.

Reply:
This is interesting - I think that DI statuses are strings and not labels, at least in part, because the values are heavily used - even primarily used - as flags in the code for conditional logic, etc. As a result translating them causes all sorts of problems - even if we consistently tested against the label value, different users can be in different languages. I definitely would be cautious about making that change for the DI labels without carefully taking into account the potential impact of translations and other label overrides. 

This seems like a similar case here, and so I might be on a starting position of preferring them declared as strings here (or perhaps enum?). But I imagine we do want translated values to appear to user - so we may need to cook up a plan to make that work.

Reply:
Also - given that the eventual destination of these values is in a picklist, could we just grab the strings here from the picklist describe information?

Reply:
We can get the list from the describe, however how do I identify the 'Open' picklist value for example without a comparison with a string 'Open'?

Reply:
I'm going to mark this "resolved" based on our conversation today - although there's a broader discussion to be had I believe we agreed not to tie it to this delivery